tchannel-core/src/main/java/com/uber/tchannel/channels/Peer.java

```diff
@@ -152,31 +150,25 @@ public Connection getConnection(ChannelId channelId) {
     }
 
     public void close() {
-        for (ChannelId id : connections.keySet()) {
-            Connection conn = connections.get(id);
-            if (conn != null) {
-                conn.close();
-            }
+        for ( Connection conn : connections.values()) {
+            conn.close();
```

Comment:
Yes. connections is ConcurrentHashMap. It cannot be null. 
Like Hashtable but unlike HashMap, this class does not allow null to be used as a key or value.
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html